File: Day20_RevWords.py
def reverseWords(self,S):
    l = []
    j = 0
    for i in range(1, len(S)):
        if(S[i] == '.'):
            l.append(S[j: i])
            j = i+1
            i += 1
    l.append(S[j:])
    s = ""
    for i in range(len(l)-1, -1, -1):
        s = s + l[i]
        s = s + '.'
    return s[0:-1]

File: Day20_RevWords.cpp
#include <bits/stdc++.h>
using namespace std;

string reverseWords(string S)
{
    vector<string> v;
    int j = 0;
    for (int i = 1; i < S.length(); i++)
    {
        if (S[i] == '.')
        {
            v.push_back(S.substr(j, i - j));
            j = i + 1;
            i++;
        }
    }
    v.push_back(S.substr(j, S.length() - j));
    string s = "";
    for (int i = v.size() - 1; i >= 0; i--)
    {
        s.append(v[i]);
        s.append(".");
    }
    s.erase(s.end() - 1);
    return s;
}

int main()
{

    return 0;
}